### ISSUE TITLE
Broken permissions for global data? 

### ISSUE THREAD

**Issue by vic-t** [cosmocode/dokuwiki-plugin-struct#573]
I created a global schema named `customer`. I display the schema in a page called `list`  in the following namespace: 

`internal:customers`

The page `list` holds the following code only: 

```
====== Customers ======

---- struct global ----
schema: customer
----
```

The group `@company` has upload permissions in the `internal:*` namespace. The permissions for `internal:customers:*` or the pages within were not defined separately. 

So, `@company` has read and write permissions on the `list` page, as verified in the ACL editor and tested on the page itself. **Still, no user in the group `@company` can edit any data in the data table.** 

In the documentation it is [stated](https://www.dokuwiki.org/plugin:struct:permissions): 

> Global data is basically always public, but editing can be limited to groups and users in the schema editor.

In my understanding this means that editing the global data is open to anyone who can see it. So, as long as a user can reach a page where a global schema is presented, he can edit it. I could restrict this by entering groups or users in the schema editor in the field `allowed editors`. 

The documentation offers [further explanation](https://www.dokuwiki.org/plugin:struct:configuration) about this field. 

> Comma separated list of users and @groups who may edit this schema's data (empty for all)
> User who want to edit Page Schemas also need at least write permission for the associated page. Superuser can always edit Struct data

This seems to confirm, for global data, what is written in the permissions section - just leave it empty if you wish for anyone to have the ability to edit it. 

I actually even tried assigning groups and users as `allowed editors` in the schema editor for this global schema but that doesn't seem to have any effect whatsoever. Only users in the admin group can edit the struct data for a global schema. 

This is tested with the current versions of DokuWiki and the struct plugin.

**Comment by annda** [comment]
This might be related to #291.

I tried to reproduce the error again, works for me in a closed wiki with authplain. I'll try investigating the code a bit more.

Meanwhile, can you post your PHP version and which authentication method you are using?

**Comment by annda** [comment]
The problem was caused by a strict ACL check on the page that the struct data is associated with (`pid`). In the case of global data, `pid` is empty and the check was performed on the root namespace. This is why the bug appeared only is some ACL setups but not others. @vic-t Thanks for the detailed report that helped me trace the source of the errors.

Fixed with 4c3a60d826c9233354d7dc5e489f4352e1971965 included in the latest release 2021-06-15

**Comment by mtkirchner** [comment]
Sorry, did not see you changing from #291.
Yes, With 2021-06-17 it works for me.
Thanks!
